master/buildbot/reporters/github.py

```diff
@@ -35,13 +35,14 @@
 from buildbot.util import httpclientservice
 from buildbot.util.giturlparse import giturlparse
 
-HOSTED_BASE_URL = 'https://api.github.com'
+HOSTED_API_URL = 'https://api.github.com'
+HOSTED_BASE_REGEX = 'github\.com\/(.+?)\/'
 
 
 class GitHubStatusPush(ReporterBase):
     name = "GitHubStatusPush"
 
-    def checkConfig(self, token, context=None, baseURL=None, verbose=False,
+    def checkConfig(self, token, context=None, apiURL=None, baseREGEX=None, verbose=False,
```

Comment:
Let's not rename baseURL here, this is API break and would need users to upgrade their configurations.

Let's name `baseREGEX` something like `base_re`. It also needs documentation.